modules/reactor/src/meshgenerators/PinMeshGenerator.C

```diff
@@ -0,0 +1,480 @@
+          duct_ids.push_back(_region_ids[0][i]);
+        }
+        else
+        {
+          background_intervals = _intervals[i];
+          background_ids.push_back(_region_ids[0][i]);
+        }
+      }
+      if (ring_intervals.size() > 0)
+      {
+        params.set<std::vector<Real>>("ring_radii") = _ring_radii;
+        if (ring_intervals.front() != 1)
+          ring_ids.insert(ring_ids.begin(), ring_ids.front());
+
+        params.set<std::vector<subdomain_id_type>>("ring_block_ids") = ring_ids;
+        params.set<std::vector<unsigned int>>("ring_intervals") = ring_intervals;
+      }
+      else if (background_intervals > 1)
+      {
+        background_ids.insert(background_ids.begin(), background_ids.front());
+      }
+
+      params.set<std::vector<subdomain_id_type>>("background_block_ids") = background_ids;
+      params.set<unsigned int>("background_intervals") = background_intervals;
+
+      if (duct_intervals.size() > 0)
+      {
+        params.set<MooseEnum>("duct_sizes_style") = "apothem";
+        params.set<std::vector<Real>>("duct_sizes") = _duct_halfpitch;
+        params.set<std::vector<subdomain_id_type>>("duct_block_ids") = duct_ids;
+        params.set<std::vector<unsigned int>>("duct_intervals") = duct_intervals;
+      }
+      params.set<MooseEnum>("polygon_size_style") = "apothem";
+      params.set<Real>("polygon_size") = _pitch / 2.0;
+
+      _build_mesh =
+          &addMeshSubgenerator("PolygonConcentricCircleMeshGenerator", name() + "_2D", params);
+    }
+
+    if (hasMeshProperty("pitch_meta", name() + "_2D"))
+      declareMeshProperty("pitch_meta", getMeshProperty<Real>("pitch_meta", name() + "_2D"));
+    if (hasMeshProperty("background_intervals_meta", name() + "_2D"))
+      declareMeshProperty(
+          "background_intervals_meta",
+          getMeshProperty<unsigned int>("background_intervals_meta", name() + "_2D"));
+    if (hasMeshProperty("node_id_background_meta", name() + "_2D"))
+      declareMeshProperty("node_id_background_meta",
+                          getMeshProperty<unsigned int>("node_id_background_meta", name() + "_2D"));
+    if (hasMeshProperty("pattern_pitch_meta", name() + "_2D"))
+      declareMeshProperty("pattern_pitch_meta",
+                          getMeshProperty<Real>("pattern_pitch_meta", name() + "_2D"));
+    if (hasMeshProperty("azimuthal_angle_meta", name() + "_2D"))
+      declareMeshProperty(
+          "azimuthal_angle_meta",
+          getMeshProperty<std::vector<Real>>("azimuthal_angle_meta", name() + "_2D"));
+    if (hasMeshProperty("num_sectors_per_side_meta", name() + "_2D"))
+      declareMeshProperty(
+          "num_sectors_per_side_meta",
+          getMeshProperty<std::vector<unsigned int>>("num_sectors_per_side_meta", name() + "_2D"));
+    if (hasMeshProperty("max_radius_meta", name() + "_2D"))
+      declareMeshProperty("max_radius_meta",
+                          getMeshProperty<Real>("max_radius_meta", name() + "_2D"));
```

Comment:
this is duplicated from above. Consolidate the two by taking it out of the if-statement

Reply:
Added this block to the end of the if-else block